### ISSUE TITLE
github auth via token

### ISSUE THREAD

**Issue by jasonkarns** [geddski/grunt-release#71]
readme says that tokens can be used, but checking the code I don't see any reference to where the auth token would be configured.

is the readme a bit premature?

**Comment by bobbyrne01** [comment]
:+1:

**Comment by scottux** [comment]
:+1:

**Comment by mpint** [comment]
This is still an issue.  And a simple fix.  If you're using auth tokens, this is the issue in grunt-release.js.
```
      request
        .post('https://api.github.com/repos/' + options.github.repo + '/releases')
        .auth(process.env[options.github.usernameVar], process.env[options.github.passwordVar])
        .set('Accept', 'application/vnd.github.manifold-preview')
        .set('User-Agent', 'grunt-release')
        .send({
          'tag_name': tagName,
          name: tagMessage,
          prerelease: type === 'prerelease'
        })
```

TL;DR - .auth(process.env[options.github.usernameVar], '') where `options.github.usernameVar` is your auth token fixes it.

The plugin expects the following to exist in your grunt task configuration

```
            github: {
                repo: 'your_name/your_repo',
                usernameVar: 'YOUR_NAME',
                passwordVar: 'YOUR_PASS'
            }
```

If you want to use a key, your grunt task may look like this

```
            github: {
                repo: 'your_name/your_repo',
                usernameVar: 'YOUR_KEY'
            }
```

Which fails because `options.github.passwordVar` isn't set in the `.auth()` method above.

To make things work until a fix, you must do the following:

```
            github: {
                repo: 'your_name/your_repo',
                usernameVar: 'YOUR_KEY',
                passwordVar: 'YOUR_PASS'
            }
```

And set the environment variable `YOUR_PASS=''`

**Comment by drublic** [comment]
We now have an option `github.accessToken` which you can set with your token. This just landed in 111eaf4a5e6d4a305196e10c0d58e157de8abc68. Release will be here today.
Thank you so much for your input and sorry for the late response.
